### ISSUE TITLE
Can't passthrough blob URLs

### ISSUE THREAD

**Issue by rassie** [miragejs/miragejs#476]
I'm working with blob URLs and can't get Mirage to passthrough them. Observer failing tests (for mirage's `master` branch):

```
  test("it can passthrough blob: URLs", async () => {
    server.config({
      routes() {
        this.passthrough();
      },
    });

    await expect(
      fetch("blob:http://localhost:9999/e8b81f12-5721-4208-a197-c9140989a7d4")
    ).rejects.toThrow("Network request failed");
  });

  test("it can passthrough blob: URLs with a function", async () => {
    server.config({
      routes() {
        this.passthrough(() => {
          return true;
        });
      },
    });

    await expect(
      fetch("blob:http://localhost:9999/e8b81f12-5721-4208-a197-c9140989a7d4")
    ).rejects.toThrow("Network request failed");
  });
```
Instead of a network error, `Mirage: Your app tried to GET 'blob:http://localhost:9999/e8b81f12-5721-4208-a197-c9140989a7d4', but there was no route defined to handle this request. Define a route for this endpoint in your routes() config. Did you forget to define a namespace?` is getting thrown.

I hope I'm not missing something major :)

**Comment by ryanto** [comment]
Yup this looks like a bug. I think the code that's creating the passthrough is getting confused when parsing the blob URL. My guess is we'll need to add some special handling when parsing these URLs.

Thanks for the failing test cases.

**Comment by rubenslopes** [comment]
I'm having the same problem with `data:image`, my work around seems like this:

```javascript
routes () {
  ...
  this.passthrough();
  this.passthrough(request => request.url.startsWith('data:image'));
}
```

**Comment by rubenslopes** [comment]
The first `passthrough()` is there because it's a ongoing project that we are implementing miragejs now

**Comment by joelsantosbr** [comment]
I have the same problem, even using the function: `this.passthrough(request => request.url.startsWith('blob')),` still giving error. The function for other cases works perfectly!

**Comment by IanVS** [comment]
I have bad news and good news.  

The bad news: there is a limitation of pretender.js (the default interceptor used by Mirage) that it only handles string and ArrayBuffer response bodies.  

The good news: Mirage now _experimentally_ supports [MSW](https://mswjs.io/), which can be used instead of pretender.  MSW allows any valid response body, and now when using Mirage 0.2.0-alpha.2+ and [mirage-msw](https://github.com/miragejs/mirage-msw) 0.2.2+, you can mock binary data as a Blob.  

`mirage-msw` is _mostly_ a drop-in replacement for pretender: 

```ts
import MSWInterceptor from 'mirage-msw';
import { createServer } from 'miragejs';

await createServer({
  interceptor: new MSWInterceptor(),
  // ... rest of your config
});
```

Note that `createServer` is now async and should be awaited (because msw is async).  You'll also need to install `msw` version 2 or higher, and run [`msw init`](https://mswjs.io/docs/integrations/browser#copy-the-worker-script) to configure it in your project.  If you decide to try it out and have any questions, feel free to post here or open an issue in the mirage-msw repo.
